hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -298,6 +247,26 @@ public StorageStatistics provide() {
         LOG.debug("Using metadata store {}, authoritative={}",
             getMetadataStore(), allowAuthoritative);
       }
+
+      ExecutorService downloadExecutorService = Executors.newFixedThreadPool(
```

Comment:
Where is this executor shutdown? Assuming we probably want the downloader to be `Closeable` and close that in `S3AFileSystem`‘s `close()` as well.

Reply:
why add a new thread pool at all?